src/_document/render.tsx

```diff
@@ -22,17 +27,36 @@ export default async function renderDocument({
     ? customDocumentFile.default
     : ((NextDocument as unknown) as DocumentType);
 
-  // TODO: this should get updated by HeadManagerContext
-  // See: https://github.com/vercel/next.js/blob/2790ea411b8b0bf2be627a5cc5ba77626079c46d/packages/next/next-server/server/render.tsx#L563
-  // Problem, can we get this to evaluate to false: https://github.com/vercel/next.js/blob/2790ea411b8b0bf2be627a5cc5ba77626079c46d/packages/next/next-server/lib/side-effect.tsx#L3?
-  const head: JSX.Element[] = [];
+  let head: JSX.Element[] = [];
 
   const renderPage: RenderPage = () => {
-    // Render markup that will be injected into #__next element by Next.JS in the Main component of Document
-    const html = renderToStaticMarkup(pageElement);
+    const html = renderToStaticMarkup(
+      <HeadManagerContext.Provider
+        value={{
+          updateHead: (state) => {
+            head = state;
+          },
+          mountedInstances: new Set(),
+        }}
+      >
+        {pageElement}
+      </HeadManagerContext.Provider>
+    );
     return { html, head };
   };
 
+  // https://github.com/vercel/next.js/blob/canary/packages/next/pages/_document.tsx#L524
```

Comment:
I'd replace this link (and any other link pointing to git repos' files) with:
https://github.com/vercel/next.js/blob/v10.0.3/packages/next/pages/_document.tsx#L524

..in order to make sure we don't lose reference to the relevant code when target repo changes. :)

Reply:
Good idea -- done;